src/stb_vorbis.h

```diff
@@ -5656,6 +5657,11 @@ int stb_vorbis_get_samples_float_interleaved(stb_vorbis *f, int channels, float
          break;
    }
    f->current_playback_loc += n;
+   if(f->current_playback_loc > lgs) {
```

Comment:
Please check for the `lgs >= 0`, otherwise, it may return 0 (on any error) or -1 when it has an invalid duration value.

Reply:
added a `&& lgs >= 0` for checking it

Reply:
Wait a minute, the return value should be positive as is an unsigned int, how could it be `-1`?

Reply:
I read the code and "1 when it has an invalid duration value" doesn't exist, it can only be 0.

Reply:
It CAN return -1, because of:
```
#define SAMPLE_unknown  0xffffffff
```
when it's a signed 32-bit integer, it's `-1`.

Reply:
There's an if before return that makes it zero, no?

Reply:
Well added an extra check for lgs, but lgs gets it's value from `stb_vorbis_stream_length_in_samples`, the `-1` can only happen in other things... This code is a bit terrible because it mixes the `SAMPLE_unknown` definition, the `-1`, the direct `0xffffffff`... And I have no idea what the value it will be once it changes to 64 bit. `UINT_MAX` looks easier to understand.

Reply:
On my end of the fork, I already replaced all the offsets from int into int64_t (otherwise it won't allow play large files, longer than 5 hours, and also depending on the sample rate), and the value of `SAMPLE_unknown` I simply replaced with `-1`: simple and easy.: https://github.com/WohlSoft/SDL-Mixer-X/blob/master/src/codecs/stb_vorbis/stb_vorbis.h

Reply:
> SAMPLE_unknown

Why not to just use the `SAMPLE_unknown` value directly?

Reply:
In your code there is a mix of int64_t and uint64_t, shouldn't it be all uint64_t ? I felt that it would be easier to match the eventual 64-bit upgrade by using an existent constant instead of the current manual 0xf...f.

Reply:
btw, I won't change anything right now because GitHub PR is melting: https://www.githubstatus.com/

Reply:
It appears GitHub is normalizing